apps/studio/src/contexts/EditorDrawerContext.tsx

```diff
@@ -55,12 +56,21 @@ export function EditorDrawerProvider({
   const [savedPageState, setSavedPageState] =
     useState<IsomerSchema>(initialPageState)
   // State of the page to render in the preview
-  const [previewPageState, setPreviewPageState] =
+  const [previewPageState, _setPreviewPageState] =
     useState<IsomerSchema>(initialPageState)
   // Holding state for images/files that have been modified in the page
   const [modifiedAssets, setModifiedAssets] = useState<ModifiedAsset[]>([])
   const [addedBlockIndex, setAddedBlockIndex] = useState<number | null>(null)
 
+  const setPreviewPageState = useCallback(
```

Comment:
might want to add a comment here saying why this is done.

Reply:
added huge paragraph!